src/utils/utils.rs

```diff
@@ -24,6 +25,25 @@ pub use crate::utils::utils::raw::{is_file, path_exists};
 #[derive(Debug, PartialEq, Eq)]
 pub struct ExitCode(pub i32);
 
+impl BitAnd for ExitCode {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        match self.0 {
```

Comment:
Maybe add a short comment explaining the conceptual logic?

```
// If `self` is `0` (success), yield `other`
```

(Also for the other impl.)

Reply:
@djc Done!